.github/workflows/codeql-analysis.yml

```diff
@@ -3,9 +3,6 @@ name: 'CodeQL'
 on:
   push:
     branches: [main]
-  pull_request:
```

Comment:
To be honest I would still run CodeQL for PRs against main, as it will block the PR from being merged if there are CodeQL issues.

Reply:
I think it's probably fine to run it on main only, because I don't remember a single time where it failed for a valid reason 😅 We'll still get the notification if it's failed on main and see the results here: https://github.com/getsentry/sentry-java/security/code-scanning

The reason we don't want to run it on PRs because without build-cache it'd be the longest one to run. Alternatively we could skip the run altogether, if there were no code changes to prevent the job from failing, but I think it's a good middle-ground for now.